contracts/ConsensusUtils.sol

```diff
@@ -110,11 +113,60 @@ contract ConsensusUtils is EternalStorage, ValidatorSet {
     }
   }
 
+  function _addToUboundingList(address _staker, uint256 amount) internal {
+    require(unboundingQueueLength(_staker) < MAX_WITHDRAW_QUEUE_LENGTH);
+    uintArrayStorage[keccak256(abi.encodePacked("unboundingQueueBlock", _staker))].push(block.number + UNBOUNDING_PERIOD);
+    uintArrayStorage[keccak256(abi.encodePacked("unboundingQueueAmount", _staker))].push(amount);
+  }
+
+  function unbound(address _staker) internal {
+    uint256 toReturn = 0;
+    uint256 currentBlock = block.number;
+    uint256 index = 0;
+    uint256 oldArrayLength = unboundingQueueLength(_staker);
+
+    //loop through queue and check if we have anything to withdraw
+    for (uint256 i = 0; i < oldArrayLength; i ++)
+    {
+      if (i > MAX_WITHDRAW_CHUNK_SIZE)
+      {
+        break;
+      }
+      if (uintArrayStorage[keccak256(abi.encodePacked("unboundingQueueBlock", _staker))][i] > currentBlock)
+      {
+        break;
+      }
+      toReturn = toReturn.add(uintArrayStorage[keccak256(abi.encodePacked("unboundingQueueAmount", _staker))][i]);
+    }
+
+    //check if we have anything
+    require(toReturn != 0);
+    index = i - 1;
+
+    //shift queue and pop back
+    for (i = 0; i < oldArrayLength - index; i++) {
+      uintArrayStorage[keccak256(abi.encodePacked("unboundingQueueBlock", _staker))][i] = uintArrayStorage[keccak256(abi.encodePacked("unboundingQueueBlock", _staker))][i + index];
```

Comment:
have you considered the gas costs for all these keccak256 calls? is there a less gas-intensive way to handle this?